addon/validations/factory.js

```diff
@@ -454,6 +462,33 @@ function getCPDependentKeysFor(attribute, validations, owner) {
 }
 
 /**
+ * Extract all dependentKeys from any property that is a CP
+ *
+ * @method extractOptionsDependentKeys
+ * @param  {Object} options
+ * @return {Array}  dependentKeys
+ */
+function extractOptionsDependentKeys(options) {
+  if(options && typeof options === 'object') {
+    return Object.keys(options).reduce((arr, key) => {
+      let option = options[key];
+
+      if(typeof option === 'object' && option.isDescriptor) {
+        let dependentKeys = option._dependentKeys || [];
```

Comment:
There could def. be use-cases when you want your validation to be dependent on another model All im doing is extracting the dependent keys from the option’s dependents so you dont have to specify it twice. So lets say I want my username validator to recompute when `foo` changes, you would have to add `foo` to your dependentKeys since it wont be in any other specified option 

`validator('length', { min: 5, dependentKeys: ['foo'] })`